aws_lambda_powertools/utilities/parser/models/s3.py

```diff
@@ -11,88 +11,319 @@
+        ],
+    )
+    versionId: Optional[str] = Field(
+        default=None,
+        description="The version ID of the object (if versioning is enabled).",
+        examples=[
+            "096fKKXTRTtl3on89fVO.nfljtsv6qko",
+            "null",
+        ],
+    )
 
 
 class S3Message(BaseModel):
-    s3SchemaVersion: str
-    configurationId: str
+    s3SchemaVersion: str = Field(
+        description="S3 schema version.",
+        examples=[
+            "1.0",
+        ],
+    )
+    configurationId: str = Field(
+        description="ID of the bucket notification configuration.",
+        examples=[
+            "828aa6fc-f7b5-4305-8584-487c791949c1",
+            "f99fa751-7860-4d65-86cb-10ff34448555",
+            "b1d3a482-96eb-4d3a-abd7-763662a6ba94",
+        ],
+    )
     bucket: S3Bucket
     object: S3Object  # noqa: A003
 
 
 class S3EventNotificationObjectModel(BaseModel):
-    key: str
-    size: Optional[NonNegativeFloat] = None
-    etag: str = Field(default="")
-    version_id: Optional[str] = Field(None, alias="version-id")
-    sequencer: Optional[str] = None
+    key: str = Field(
+        description="The object key (file name) of the S3 object.",
+        examples=[
+            "my-image.jpg",
+            "documents/report.pdf",
+            "logs/2023/01/15/app.log",
+        ],
+    )
+    size: Optional[NonNegativeFloat] = Field(
+        default=None,
+        description="The size of the object in bytes.",
+        examples=[1024, 2048576, 0],
+    )
+    etag: str = Field(
+        default="",
+        description="The entity tag (ETag) of the object.",
+        examples=[
+            "d41d8cd98f00b204e9800998ecf8427e",
+            "098f6bcd4621d373cade4e832627b4f6",
+        ],
+    )
+    version_id: Optional[str] = Field(
+        default=None,
+        alias="version-id",
+        description="The version ID of the object (if versioning is enabled).",
+        examples=[
+            "096fKKXTRTtl3on89fVO.nfljtsv6qko",
+            "null",
+        ],
+    )
+    sequencer: Optional[str] = Field(
+        default=None,
+        description="A string representation of a hexadecimal value used to determine event sequence.",
+        examples=[
+            "0A1B2C3D4E5F678901",
+            "005B21C13A6F24045E",
+        ],
+    )
 
 
 class S3EventNotificationEventBridgeBucketModel(BaseModel):
-    name: str
+    name: str = Field(
+        description="Name of the Amazon S3 bucket.",
+        examples=[
+            "lambda-artifacts-deafc19498e3f2df",
+            "example-bucket",
+            "sourcebucket",
+        ],
+    )
 
 
 class S3EventNotificationEventBridgeDetailModel(BaseModel):
-    version: str
+    version: str = Field(
+        description="Version of the event message. Currently 0 (zero) for all events",
+        examples=[
+            "0",
+        ],
+    )
     bucket: S3EventNotificationEventBridgeBucketModel
     object: S3EventNotificationObjectModel  # noqa: A003
```

Comment:
Can you add a description here? Please let me know if you need help writing them.